Python/Product/Analysis/Values/ProtocolInfo.cs

```diff
@@ -284,33 +285,42 @@ class ProtocolInfo : AnalysisValue, IHasRichDescription {
         }
 
         public virtual IEnumerable<KeyValuePair<string, string>> GetRichDescription() {
-            var name = _protocols.OfType<NameProtocol>().FirstOrDefault();
+            var names = _protocols.OfType<NameProtocol>().ToArray();
+            Debug.Assert(names.Length <= 1, "Multiple names are not supported");
+            var name = names.FirstOrDefault();
             if (name != null) {
-                yield return new KeyValuePair<string, string>(WellKnownRichDescriptionKinds.Type, name.Name);
+                return name.GetRichDescription();
             }
-            if (_protocols.Any()) {
-                bool firstLoop = true;
-                foreach (var p in _protocols.Where(pr => !(pr is NameProtocol))) {
-                    if (firstLoop) {
-                        firstLoop = false;
-                        if (name != null) {
-                            yield return new KeyValuePair<string, string>(WellKnownRichDescriptionKinds.Misc, "(");
-                        }
-                    } else {
-                        yield return new KeyValuePair<string, string>(WellKnownRichDescriptionKinds.Comma, ", ");
-                    }
-                    foreach (var d in p.GetRichDescription()) {
-                        yield return d;
+
+            var res = new List<KeyValuePair<string, string>>();
+            var namespaces = _protocols.OfType<NamespaceProtocol>().ToArray();
+            var other = _protocols.Except(names).Except(namespaces).ToArray();
+
+            var fallbackName = other.Select(p => p.Name).FirstOrDefault(n => !string.IsNullOrEmpty(n)) ?? "<unknown>";
+            if (!string.IsNullOrEmpty(fallbackName)) {
```

Comment:
It looks like `fallbackName` can't be null or empty

Reply:
Highly unlikely, true, but this will cover any future changes that may leave it empty.